### ISSUE TITLE
Changes in any file starting with underscore triggers recompilation of all files

### ISSUE THREAD

**Issue by sKopheK** [glenn2223/vscode-live-sass-compiler#17]
...even when the file starting with underscore is not imported to any SCSS file, its changes trigger recomplication of all SCSS files

1) turn watch on, open output tab
2) create file `_x.scss`
3) edit and save the file
4) output window prints out
```
Change detected...
_x.scss
--------------------
Compiling Sass/Scss Files: 
...
```

**Comment by glenn2223** [comment]
Hi @sKopheK,

This was by design. When Ritwick Dey created the extension it was built upon the generic principle of if it's a partial it could be included. So lets keep it simple and compile every file (settings dependant).

There was an issue raised over in [his repo](https://github.com/ritwickdey/vscode-live-sass-compiler) that covered it. From what I remember it was decided that there wouldn't be a simple way of doing it without effecting the performance of the extension itself. However, for the life of me, I can't find the issue!

Closing as `wontfix`. However, if this becomes a real issue across multiple users then I will reconsider taking the time to do a full workaround on the extension.

👍 if you want this adding

**Comment by sKopheK** [comment]
thanks for the answer. switching from NetBeans IDE, I'm used to compiling only what was really changed - however there was still sass@ruby used and thus it was really slow, especially when recompiling all files. pitty, i won't be getting much performance boost when updating any of my many partials :(

**Comment by sKopheK** [comment]
project with ~400 files suffers a lot by this restriction, please take a look on the video:
https://drive.google.com/file/d/1vO4G3DcLzu3UPCvMwDEgu6bypuGzTiQc/view?usp=sharing
also added you to a private repo with the files shown in the video so you can test yourself.

when other files such as JS, PHP or HTML is there in the project folder, it gets even more slow because it waits until github scanning is done, which when it kicks in it goes into some kind of pending mode and watch is delayed by a couple of tens of seconds, but this might be an issue of git integration in the VS Code: https://drive.google.com/file/d/10cS_sAn-16MYWuVEMiwd8pSD6qaJBCu6/view?usp=sharing

**Comment by glenn2223** [comment]
Yeah, I've seen and can feel your pain. I'll see if I can find anything to implement that'll help.

It'll be slow to research, check, implement and test as I'm a bit rammed at the moment with work, uni study, etc, etc. It's not stopping me from investing time into my opensource stuff, but it's definitely reduced it compared to Christmas (when everything else stopped 😄 )

**Comment by glenn2223** [comment]
Doing some testing, I think the main delay comes from git integration (as you suggest).  
This is because the initial pull back of files is instant on my machine (when the `Compiling Sass/Scss Files:` output is shown).  
However compiling does take a few seconds to complete for all the files (when the output of `Watching...` is shown).

**I can see the issue this causes** when compiling everything, even those files that don't actually use the partial.

Initial thoughts: compile a dictionary of file data (the file, the imports, the date it was last modified) and updating checking it as files save, are created and on launch. 

v5 coming soon so this may be a v6 or v5.x (depending on the possibility of breaking changes, 🤞 for none)

Leaving closed until after v5 release

**Comment by sKopheK** [comment]
Hi, a little update on this one - was using NetBeans since its support for changed files solves the partials issue mentioned above, but as it runs sass compiler for every single changed file, it's also slow. Running the sass executable directly with `--watch` parameter is blazing fast tho and the  `--stop-on-error` parameter is something you might want to add support for in here as well.

https://sass-lang.com/documentation/cli/dart-sass

**Comment by enoversum** [comment]
Chiming in here as well. Only v5 did this freaking compilation of files starting with an underscore, it had been carefully ignored beforehand. I would also love to have the old handling back, so the _partial.scss files are not being compiled to the various formats.

**Comment by glenn2223** [comment]
> Chiming in here as well. Only v5 did this freaking compilation of files starting with an underscore, it had been carefully ignored beforehand. I would also love to have the old handling back, so the _partial.scss files are not being compiled to the various formats.

I wasn't able to replicate this. Partials have always triggered the compilation of all files unless excluded by your settings.

Can you open a new issue with details on how to replicate it? I can then take a look tomorrow.

*FYI: this particular issue relates to a future change. One where the extension knows what files include the partial file that has been saved and only compiles those dependant files*

**Comment by enoversum** [comment]
> > Chiming in here as well. Only v5 did this freaking compilation of files starting with an underscore, it had been carefully ignored beforehand. I would also love to have the old handling back, so the _partial.scss files are not being compiled to the various formats.
> 
> I wasn't able to replicate this. Partials have always triggered the compilation of all files unless excluded by your settings.
> 
> Can you open a new issue with details on how to replicate it? I can then take a look tomorrow.
> 
> _FYI: this particular issue relates to a future change. One where the extension knows what files include the partial file that has been saved and only compiles those dependant files_

Well, I didn't change anything on my side. I have a styles.scss file that imports several files starting with an underscore, like _basics.scss. If I edited something in them and saved them, the compiler normally ignored it and only compiled the full styles.scss file once I saved that, but didn't create e.g. basics.css, basics.min.css. I can tell, because I work with my tiny framework all the time and didn't have this until rc1 of version 5. I'll create another issue for that, agree :) .

**Comment by glenn2223** [comment]
> Well, I didn't change anything on my side. I have a styles.scss file that imports several files starting with an underscore, like _basics.scss. If I edited something in them and saved them, the compiler normally ignored it and only compiled the full styles.scss file once I saved that, but didn't create e.g. basics.css, basics.min.css. I can tell, because I work with my tiny framework all the time and didn't have this until rc1 of version 5. I'll create another issue for that, agree :) .

If you could open a new issue that would be great. Please also include an example file structure (something like the below) and your current settings. Thanks

```
-- Sass
  -- _partial.scss
  -- SubFolder
    -- File.scss
-- Css
  -- Subfolder
    -- File.min.css
```

**Comment by enoversum** [comment]
> > Well, I didn't change anything on my side. I have a styles.scss file that imports several files starting with an underscore, like _basics.scss. If I edited something in them and saved them, the compiler normally ignored it and only compiled the full styles.scss file once I saved that, but didn't create e.g. basics.css, basics.min.css. I can tell, because I work with my tiny framework all the time and didn't have this until rc1 of version 5. I'll create another issue for that, agree :) .
> 
> If you could open a new issue that would be great. Please also include an example file structure (something like the below) and your current settings. Thanks
> 
> ```
> -- Sass
>   -- _partial.scss
>   -- SubFolder
>     -- File.scss
> -- Css
>   -- Subfolder
>     -- File.min.css
> ```

Okay, already created an issue, will try to make it a bit more specific.

**Comment by virik** [comment]
I am experiencing some issues with this as well. I have read the above comments, and I understand why all files are generated. However I have multiple projects in my workspace, and saving a partial also generates files for my other projects. This might cause problems if other developers have working on the projects in the meantime. Would it be possible to have the extension only generate files for the related project (project root folder). Example: https://d.pr/i/qnAV4D
